core/node/shared_metrics/build.rs

```diff
@@ -4,43 +4,75 @@ use std::{
     env, fs,
     io::{self, Write},
     path::Path,
+    process::Command,
 };
 
 use rustc_version::{Channel, LlvmVersion};
 
-fn print_rust_meta(out: &mut impl Write, meta: &rustc_version::VersionMeta) -> io::Result<()> {
+fn print_binary_meta(out: &mut impl Write) -> io::Result<()> {
+    let rustc_meta = rustc_version::version_meta().expect("Failed obtaining rustc metadata");
+
     writeln!(
         out,
-        "pub(crate) const RUSTC_METADATA: RustcMetadata = RustcMetadata {{ \
-            version: {semver:?}, \
-            commit_hash: {commit_hash:?}, \
-            commit_date: {commit_date:?}, \
-            channel: {channel:?}, \
+        "pub const BIN_METADATA: BinMetadata = BinMetadata {{ \
+            rustc_version: {semver:?}, \
+            rustc_commit_hash: {commit_hash:?}, \
+            rustc_commit_date: {commit_date:?}, \
+            rustc_channel: {channel:?}, \
             host: {host:?}, \
-            llvm: {llvm:?} \
+            llvm: {llvm:?}, \
+            git_branch: {git_branch:?}, \
+            git_revision: {git_revision:?} \
         }};",
-        semver = meta.semver.to_string(),
-        commit_hash = meta.commit_hash,
-        commit_date = meta.commit_date,
-        channel = match meta.channel {
+        semver = rustc_meta.semver.to_string(),
+        commit_hash = rustc_meta.commit_hash,
+        commit_date = rustc_meta.commit_date,
+        channel = match rustc_meta.channel {
             Channel::Dev => "dev",
             Channel::Beta => "beta",
             Channel::Nightly => "nightly",
             Channel::Stable => "stable",
         },
-        host = meta.host,
-        llvm = meta.llvm_version.as_ref().map(LlvmVersion::to_string),
+        host = rustc_meta.host,
+        llvm = rustc_meta.llvm_version.as_ref().map(LlvmVersion::to_string),
+        git_branch = git_branch(),
+        git_revision = git_revision()
     )
 }
 
+/// Outputs the current git branch as a string literal.
+pub fn git_branch() -> String {
+    run_cmd("git", &["rev-parse", "--abbrev-ref", "HEAD"])
+}
+
+/// Outputs the current git commit hash as a string literal.
+pub fn git_revision() -> String {
+    run_cmd("git", &["rev-parse", "--short", "HEAD"])
+}
+
+/// Tries to run the command, only returns `Some` if the command
+/// succeeded and the output was valid utf8.
+fn run_cmd(cmd: &str, args: &[&str]) -> String {
+    run_cmd_opt(cmd, args).unwrap_or("unknown".to_string())
```

Comment:
Bikeshedding: Is "unknown" a better value than no value at all (i.e., `None`)? IMO, the latter is more idiomatic, and it fits better at least for metrics.

Reply:
Addressed in https://github.com/matter-labs/zksync-era/pull/3193/commits/6bc90e2f8364418fb8632548d59e397bf94ca29d